examples/3d/custom_render.rs

```diff
@@ -0,0 +1,291 @@
+) {
+    // Create a new shader pipeline
+    let pipeline_handle = pipelines.add(PipelineDescriptor::default_config(ShaderStages {
+        vertex: shaders.add(Shader::from_glsl(ShaderStage::Vertex, VERTEX_SHADER)),
+        fragment: Some(shaders.add(Shader::from_glsl(ShaderStage::Fragment, FRAGMENT_SHADER))),
+    }));
+
+    // Add an AssetRenderResourcesNode to our Render Graph. This will bind MyMaterial resources to our shader
+    render_graph.add_system_node(
+        "my_material",
+        AssetRenderResourcesNode::<MyMaterial>::new(true),
+    );
+
+    // Add a MyMeshNode to our Render Graph. This will allow us to bind custom vertices to our shader.
+    render_graph.add_system_node("my_mesh", MyMeshNode::default());
+
+    // Add a Render Graph edge connecting our new "my_material" node to the main pass node. This ensures "my_material" runs before the main pass
+    render_graph
+        .add_node_edge("my_material", base::node::MAIN_PASS)
+        .unwrap();
+
+    // Add a Render Graph edge connecting our new "my_mesh" node to the main pass node. This ensures "my_mesh" runs before the main pass
+    render_graph
+        .add_node_edge("my_mesh", base::node::MAIN_PASS)
+        .unwrap();
+
+    // Create a new material
+    let material = materials.add(MyMaterial {
+        color: Color::rgb(0.0, 0.8, 0.0),
+    });
+
+    let mesh = meshes.add(MyMesh {});
+
+    // Setup our world
+    commands
+        // cube
+        .spawn(MeshComponents {
+            render_pipelines: RenderPipelines::from_pipelines(vec![RenderPipeline::specialized(
+                pipeline_handle,
+                // NOTE: in the future you wont need to manually declare dynamic bindings
+                PipelineSpecialization {
+                    dynamic_bindings: vec![
+                        // Transform
+                        DynamicBinding {
+                            bind_group: 1,
+                            binding: 0,
+                        },
+                        // MyMaterial_color
+                        DynamicBinding {
+                            bind_group: 1,
+                            binding: 1,
+                        },
+                    ],
+                    ..Default::default()
+                },
+            )]),
+            translation: Translation::new(0.0, 0.0, 0.0),
+            ..Default::default()
+        })
+        .with(material)
+        .with(mesh)
+        // camera
+        .spawn(Camera3dComponents {
+            transform: Transform::new_sync_disabled(Mat4::face_toward(
+                Vec3::new(3.0, 5.0, -8.0),
+                Vec3::new(0.0, 0.0, 0.0),
+                Vec3::new(0.0, 1.0, 0.0),
+            )),
+            ..Default::default()
+        });
+}
+
+pub fn my_mesh_node_system(
+    mut state: Local<MyMeshNodeState>,
+    render_resource_context: Res<Box<dyn RenderResourceContext>>,
+    mut render_resource_bindings: ResMut<RenderResourceBindings>,
```

Comment:
One thing that strikes immediately is that this is unused.

Reply:
This example is incomplete, I'll be doing lots of cleanup once it get it working.